src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/ScreenRecorder.cs

```diff
@@ -0,0 +1,566 @@
+using System;
+using System.Collections;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Converters;
+using StateRecorder.JsonConverters;
+using Unity.Collections;
+using UnityEngine;
+using UnityEngine.Experimental.Rendering;
+using ErrorEventArgs = Newtonsoft.Json.Serialization.ErrorEventArgs;
+
+#if UNITY_EDITOR
+using RegressionGames;
+using UnityEditor;
+using UnityEditor.Media;
+#endif
+
+
+namespace StateRecorder
+{
+    [JsonConverter(typeof(StringEnumConverter))]
+    public enum KeyFrameReason
+    {
+        FirstFrame,
+        Keyboard,
+        Mouse,
+        SceneObject,
+        UIObject
+    }
+
+    [Serializable]
+    [SuppressMessage("ReSharper", "InconsistentNaming")]
+    public abstract class InputActionData
+    {
+        public double startTime;
+    }
+
+    [Serializable]
+    [SuppressMessage("ReSharper", "InconsistentNaming")]
+    public class FrameStateData
+    {
+        public long tickNumber;
+        public KeyFrameReason? keyFrame;
+        public float time;
+        public int[] screenSize;
+        public PerformanceMetricData performance;
+        public List<RenderableGameObjectState> state;
+        public InputData inputs;
+    }
+
+    [Serializable]
+    [SuppressMessage("ReSharper", "InconsistentNaming")]
+    public class InputData
+    {
+        public List<InputActionData> keyboard;
+        public List<InputActionData> mouse;
+    }
+
+    [Serializable]
+    [SuppressMessage("ReSharper", "InconsistentNaming")]
+    public class PerformanceMetricData
+    {
+        public float previousTickTime;
+        public int framesSincePreviousTick;
+        public int fps;
+        public EngineStatsData engineStats;
+    }
+
+    [Serializable]
+    [SuppressMessage("ReSharper", "InconsistentNaming")]
+    public class EngineStatsData
+    {
+        public float frameTime;
+        public float renderTime;
+
+        public int triangles;
+        public int vertices;
+
+        public int setPassCalls;
+
+        public int drawCalls;
+        public int dynamicBatchedDrawCalls;
+        public int staticBatchedDrawCalls;
+        public int instancedBatchedDrawCalls;
+
+        public int batches;
+        public int dynamicBatches;
+        public int staticBatches;
+        public int instancedBatches;
+    }
+
+    public class ScreenRecorder : MonoBehaviour
+    {
+        //TODO: ? Move these values to RGSettings ?
+        [Tooltip("FPS at which to capture frames")]
+        public int recordingFPS = 5;
+
+        [Tooltip("Forces image (non video) mode even in the Editor")]
+        public bool forceImageMode;
+
+        public string stateRecordingsDirectory = "/Users/zack/unity_videos";
```

Comment:
Should this be updated before merging?

Reply:
yes, I'll see if I can set it to something in the user's home directory (OS agnostic).  The unity game data directory is in such an obscure location on most platforms as to be un-useable/un-reasonable.